### PR TITLE
Refactor: Remove deprecated durable objects and update imports

### PR DESCRIPTION
**Pull request by a-kuz**
- Deleted `refresh-token.ts` and `user-messaging.ts` durable objects as they are no longer used.
- Updated import paths in `index.ts` and `user-messaging-service.ts` to reflect the new locations and names of durable objects.

This change cleans up the codebase by removing outdated components and ensures that the remaining imports point to the correct modules.

### CODE REVIEW

**Comment by a-kuz** [comment]
## Code Review Agent Run Status

- **AI Based Review:** Successful
## Code Review Overview

- **Summary:** This PR focuses on refactoring the durable objects and updating import paths, alongside introducing error handling and socket management functions. The changes aim to reorganize the code structure, improve modularity, and ensure consistent error responses. The addition of new utility functions and the restructuring of import paths reflect an effort to enhance code maintainability and scalability.
- **Code change type:** Refactoring, Feature Addition, Configuration Changes
- **Unit tests added:** False
- **[Estimated effort to review ](https://docs.bito.ai/bito-dev-agents/ai-code-review-agent/faqs#what-is-estimated-effort-to-review-in-code-review-output)(1-5, lower is better):** 2, The changes are well-structured and follow a clear pattern, making the review process straightforward. However, the addition of new functionalities and significant refactoring requires careful consideration to ensure compatibility and maintainability.




 <b> [>>See detailed code suggestions<<](https://github.com/a-kuz/big-auth/pull/7/files) </b><br>The Bito AI Code Review Agent successfully reviewed 4 files and discovered 16 issues. Please review these issues along with suggested fixes in the [Changed Files](https://github.com/a-kuz/big-auth/pull/7/files).
## High-level Feedback

 The PR introduces several key improvements and refactoring that enhance the code's organization and scalability. The removal of deprecated durable objects and the update of import paths are positive steps towards maintaining a clean codebase. Additionally, the introduction of error handling and socket management functions demonstrates an effort to improve the robustness and functionality of the code. However, the absence of unit tests for the new functionalities is a missed opportunity to ensure their correctness and reliability. Incorporating tests would significantly enhance the quality and maintainability of the code.

File: src/durable-objects/messaging/UserMessagingDO.ts [hunk 1]
```diff
@@ -1,25 +1,29 @@
-import { ChatList, ChatListItem } from "../types/ChatList";
-import { Env } from "../types/Env";
-import { BaseEvent, EditMessageEvent, NewMessageEvent } from "../types/Event";
+import { ChatList, ChatListItem } from "../../types/ChatList";
+import { Env } from "../../types/Env";
+import { EditMessageEvent, NewMessageEvent } from "../../types/Event";
+import { errorResponse } from "../../utils/error-response";
+import { internalSocket } from "./internal-socket";
+import { userSocket } from "./user-socket";
```
Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:6]: **Suggestion**: Consider adding unit tests for the newly introduced functions 'internalSocket' and 'userSocket' to ensure they behave as expected under various scenarios. Testing these functions can help identify potential issues and edge cases early in the development process. <br> **Code Suggestion**: 
 ```
 +// TODO: Implement unit tests for internalSocket and userSocket functions
 ```

File: src/durable-objects/messaging/UserMessagingDO.ts [hunk 2]
```diff
@@ -181,57 +185,65 @@ export class UserMessagingDO implements DurableObject {
...
-  async websocket(request: Request) {
-    // Expect to receive a WebSocket Upgrade request.
-    // If there is one, accept the request and return a WebSocket Response.
+  async userSocket(request: Request): Promise<Response> {
     const upgradeHeader = request.headers.get("Upgrade");
     if (!upgradeHeader || upgradeHeader !== "websocket") {
-      return new Response("Durable Object expected Upgrade: websocket", {
-        status: 426,
-      });
+      return errorResponse("Durable Object expected Upgrade: websocket", 426);
```
Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:191]: **Suggestion**: Ensure consistent usage of the 'errorResponse' utility function across all error scenarios to maintain uniform error handling behavior. Review other parts of the codebase for error handling consistency. <br> **Code Suggestion**: 
 ```
 +      // Review and ensure consistent error handling
 ```

File: src/durable-objects/messaging/UserMessagingDO.ts [hunk 3]
```diff
@@ -181,57 +185,65 @@ export class UserMessagingDO implements DurableObject {
...
+          await receiverDO.fetch(
+            new Request(`${origin}/${this.userId}/internal-websocket`),
+          )
+        ).webSocket!;
+
+        socket.send(`${this.userId} online`);
+        this.sockets.set(chat.id, socket);
+      }
+    }
+  }
```
Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:242]: **Suggestion**: The 'online' method could benefit from error handling to manage scenarios where fetching chat lists or sockets might fail. Consider wrapping the logic in a try-catch block. <br> **Code Suggestion**: 
 ```
 +  async online() {
 +    try {
 +      // Existing logic
 +    } catch (error) {
 +      console.error("Failed to execute online method:", error);
 +    }
 +  }
 ```

File: src/durable-objects/messaging/internal-socket.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  // If the client closes the connection, the runtime will close the connection too.
+  server.addEventListener("close", (cls: CloseEvent) => {
+    // messaging.offline();
+		server.send(`${userId} offline`);
+
+    server.close(cls.code);
+  });
+  return client;
+};
```
Comment (a-kuz) [inline src/durable-objects/messaging/internal-socket.ts:39]: **Suggestion**: Given the critical role of sockets in messaging, it's important to ensure robust error handling within the 'internalSocket' function. Consider adding error handling mechanisms to manage and log potential failures during socket operations. <br> **Code Suggestion**: 
 ```
 +export const internalSocket = (
 +  try {
 +    // Existing logic
 +  } catch (error) {
 +    console.error("Internal socket error:", error);
 +    throw error; // Or handle error as appropriate
 +  }
 +);
 ```

File: src/index.ts [hunk 5]
```diff
@@ -20,8 +20,8 @@ import { NetworkInfoHandler } from "./handlers/NetworkInfoHandler";
 import { GetChatsHandler } from "./handlers/GetChatsHandler";
 import { GetProfileHandler } from "./handlers/GetProfileHandler";
 
-export { RefreshTokenDO } from "./durable-objects/refresh-token";
-export { UserMessagingDO } from "./durable-objects/user-messaging";
+export { RefreshTokenDO } from "./durable-objects/RefreshTokenDO";
+export { UserMessagingDO } from "./durable-objects/messaging";
```
Comment (a-kuz) [inline src/index.ts:24]: **Suggestion**: Verify that the updated import paths in 'index.ts' correctly reflect the new locations of 'RefreshTokenDO' and 'UserMessagingDO'. Incorrect paths can lead to runtime errors or module resolution issues. <br> **Code Suggestion**: 
 ```
 +// Confirm the accuracy of the updated import paths
 ```

File: src/services/user-messaging-service.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { UserMessagingDO } from "../durable-objects/user-messaging";
+import { UserMessagingDO } from "../durable-objects/messaging/UserMessagingDO";
```
Comment (a-kuz) [inline src/services/user-messaging-service.ts:1]: **Suggestion**: After updating the import path for 'UserMessagingDO', ensure that all references to this durable object in the service layer are consistent with the new path to prevent import errors. <br> **Code Suggestion**: 
 ```
 +// Double-check all references to UserMessagingDO for consistency
 ```

File: src/durable-objects/messaging/UserMessagingDO.ts [hunk 7]
```diff
@@ -181,57 +185,65 @@ export class UserMessagingDO implements DurableObject {
...
+
+  async offline() {}
+
+  userId = "";
+  #origin = "";
+  sockets: Map<string, WebSocket> = new Map();
```
Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:248]: **Suggestion**: The introduction of public class properties 'userId', '#origin', and 'sockets' at the end of the file seems to be a refactor from private to a mix of public and private access. Ensure that the access levels are correctly set according to the intended encapsulation and usage within the class. Consider if 'userId' should also be private. <br> **Code Suggestion**: 
 ```
 +  #userId = "";
 +  #origin = "";
 +  #sockets: Map<string, WebSocket> = new Map();
 ```

File: src/index.ts (hunk 5 above)
Comment (a-kuz) [inline src/index.ts:24]: **Code Structure Issue**: The export statements for 'RefreshTokenDO' and 'UserMessagingDO' have been updated to reflect new file paths. This change is critical as it affects how these modules are imported and used throughout the application. <br> **Fix**: Verify that all import statements throughout the application that rely on 'RefreshTokenDO' and 'UserMessagingDO' are updated to match these new export paths. This ensures that the application can correctly locate and use these modules without encountering module not found errors. <br> **Code Suggestion**: 
 ```
 -export { RefreshTokenDO } from "./durable-objects/refresh-token";
 -export { UserMessagingDO } from "./durable-objects/user-messaging";
 +export { RefreshTokenDO } from "./durable-objects/RefreshTokenDO";
 +export { UserMessagingDO } from "./durable-objects/messaging";
 ```

File: src/services/user-messaging-service.ts (hunk 6 above)
Comment (a-kuz) [inline src/services/user-messaging-service.ts:1]: **Code Structure Issue**: The import path for 'UserMessagingDO' has been updated to reflect the new location of the 'UserMessagingDO' class within the project structure. This change is crucial for ensuring that the 'UserMessagingService' can correctly interact with the 'UserMessagingDO' Durable Object. <br> **Fix**: Confirm that the new import path correctly points to the 'UserMessagingDO' class's new location. Additionally, ensure that any other references to 'UserMessagingDO' within the service or elsewhere in the application are updated to reflect this new path. <br> **Code Suggestion**: 
 ```
 -import { UserMessagingDO } from "../durable-objects/user-messaging";
 +import { UserMessagingDO } from "../durable-objects/messaging/UserMessagingDO";
 ```

File: src/durable-objects/messaging/UserMessagingDO.ts (hunk 1 above)
Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:6]: **Code Structure Issue**: The import paths for ChatList, ChatListItem, Env, EditMessageEvent, and NewMessageEvent have been updated to reflect a new directory structure. Additionally, new imports for errorResponse, internalSocket, and userSocket have been added. This change suggests a significant reorganization of the codebase, which could impact how these modules interact with each other. <br> **Fix**: Ensure that all references to these imports throughout the codebase are updated to reflect the new paths. Additionally, verify that the newly imported modules (errorResponse, internalSocket, and userSocket) are being used correctly within UserMessagingDO and that their implementations are consistent with the rest of the codebase's standards and practices. <br> **Code Suggestion**: 
 ```
 -import { ChatList, ChatListItem } from "../types/ChatList";
 -import { Env } from "../types/Env";
 -import { BaseEvent, EditMessageEvent, NewMessageEvent } from "../types/Event";
 +import { ChatList, ChatListItem } from "../../types/ChatList";
 +import { Env } from "../../types/Env";
 +import { EditMessageEvent, NewMessageEvent } from "../../types/Event";
 +import { errorResponse } from "../../utils/error-response";
 +import { internalSocket } from "./internal-socket";
 +import { userSocket } from "./user-socket";
 ```

File: src/durable-objects/messaging/UserMessagingDO.ts (hunk 3 above)
Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:242]: **Performance Issue**: The method 'online' in UserMessagingDO.ts inefficiently iterates over chats to update their status to online, potentially performing unnecessary operations and network calls. This could lead to increased latency and resource consumption, especially with a large number of chats. <br> **Fix**: Implement a more efficient way to handle online status updates, possibly by tracking only active chats or by optimizing storage access patterns to reduce the overhead of fetching and updating chat statuses. <br> **Code Suggestion**: 
 ```
 -    private online: boolean,
 +    private activeChats: Set<string>,

 +  async updateChatStatusToOnline(chatId: string) {
 +    this.activeChats.add(chatId);
 +    // Additional logic to optimize online status update
 +  }
 ```

File: src/durable-objects/messaging/UserMessagingDO.ts [hunk 8]
```diff
@@ -1,25 +1,29 @@
...
 
     switch (action) {
       case "websocket":
-        return this.websocket(request);
+        return this.userSocket(request);
+      case "internal-websocket":
+        return this.internalSocket(request);
       case "send":
         return this.dialogMessage(request);
       case "edit":
```
Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:29]: **Performance Issue**: The refactoring of import paths and the addition of new imports increase the dependency on external modules, which might not be optimized and could affect the performance of the Durable Object. Especially, the dynamic import of utility functions like 'errorResponse', 'internalSocket', and 'userSocket' could introduce runtime overhead and latency. <br> **Fix**: Evaluate the necessity of each import and consider implementing critical functionality directly within the Durable Object to reduce external dependencies. For utility functions, assess if they can be simplified or inlined. <br> **Code Suggestion**: 
 ```
 -import { ChatList, ChatListItem } from "../types/ChatList";
 -import { Env } from "../types/Env";
 -import { BaseEvent, EditMessageEvent, NewMessageEvent } from "../types/Event";
 +import { ChatList, ChatListItem } from "../../types/ChatList";
 +import { Env } from "../../types/Env";
 +import { EditMessageEvent, NewMessageEvent } from "../../types/Event";
 +import { errorResponse } from "../../utils/error-response";
 ```

Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:29]: **Scalability Issue**: The refactoring introduces additional imports from relative paths that go two levels up, which could potentially lead to a fragile codebase that is harder to maintain and scale. This is especially true if the project structure changes, as updating these paths can be error-prone and tedious. <br> **Fix**: Consider using a module alias or a dependency injection framework to manage dependencies more cleanly. This approach can make the codebase more resilient to changes in the project structure and reduce the maintenance overhead associated with deep relative imports. <br> **Code Suggestion**: 
 ```
 -import { ChatList, ChatListItem } from "../types/ChatList";
 -import { Env } from "../types/Env";
 -import { BaseEvent, EditMessageEvent, NewMessageEvent } from "../types/Event";
 +import { ChatList, ChatListItem } from "../../types/ChatList";
 +import { Env } from "../../types/Env";
 +import { EditMessageEvent, NewMessageEvent } from "../../types/Event";
 +import { errorResponse } from "../../utils/error-response";

 +  async efficientChatStateUpdate(chatId: string) {
 +    // Implement efficient chat state update logic here
 +  }
 ```

File: src/durable-objects/messaging/UserMessagingDO.ts (hunk 3 above)
Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:242]: **Optimization Issue**: The method 'online' uses a for-loop to iterate over 'chatList' and potentially establish a WebSocket connection for each chat item. This approach can lead to performance issues, especially if the 'chatList' contains a large number of items. Establishing WebSocket connections is an I/O operation that can be expensive and slow, particularly when done in a synchronous loop. This can lead to increased latency in the 'online' method, impacting the responsiveness of the application. <br> **Fix**: To optimize the performance of the 'online' method, consider leveraging concurrency in establishing WebSocket connections. Instead of a synchronous for-loop, use 'Promise.all' to initiate all WebSocket connections in parallel. This approach can significantly reduce the time it takes to establish connections by taking advantage of asynchronous I/O operations. <br> **Code Suggestion**: 
 ```
 -  for (const chat of chatList) {
 +  await Promise.all(chatList.map(async (chat) => {
 +    // logic to establish WebSocket connection
 +  }));
 ```

File: src/durable-objects/messaging/UserMessagingDO.ts (hunk 2 above)
Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:191]: **Security Issue**: The error response for an incorrect WebSocket upgrade request leaks internal implementation details which could aid an attacker in crafting further attacks. Specifically, stating "Durable Object expected Upgrade: websocket" reveals the expected header and object type. <br> **Fix**: Modify the error response to be more generic and less revealing about the server's internal expectations or configurations. <br> **Code Suggestion**: 
 ```
 -      return new Response("Durable Object expected Upgrade: websocket", {
 -        status: 426,
 -      });
 +      return errorResponse("Durable Object expected Upgrade: websocket", 426);
 ```

File: src/durable-objects/messaging/UserMessagingDO.ts (hunk 8 above)
Comment (a-kuz) [inline src/durable-objects/messaging/UserMessagingDO.ts:29]: **Security Issue**: The refactor introduces new imports from relative paths that have changed, which could potentially break if the directory structure changes again. While not a direct security issue, it's a practice that could lead to maintenance issues or runtime errors if not handled properly. <br> **Fix**: Consider using a module alias or a more stable path management strategy to reduce the risk of future breakages. <br> **Code Suggestion**: 
 ```
 -import { ChatList, ChatListItem } from "../types/ChatList";
 -import { Env } from "../types/Env";
 -import { BaseEvent, EditMessageEvent, NewMessageEvent } from "../types/Event";
 +import { ChatList, ChatListItem } from "../../types/ChatList";
 +import { Env } from "../../types/Env";
 +import { EditMessageEvent, NewMessageEvent } from "../../types/Event";
 +import { errorResponse } from "../../utils/error-response";

 -  #userId = "";
 +  userId = "";
 +  #origin = "";
 ```
